project/src/main/webapp/profile.js

```diff
@@ -24,43 +29,61 @@ function updateProfileBasics(userData) {
 }
 
 /** Populate the user profile with all associated events */
-function updateProfileEvents() {
-  updateUserEventsHosting();
+async function updateProfileEvents() {
   updateUserEventsParticipating();
   updateUserEventsVolunteering();
+  await updateUserEventsHosting();
+  setDefaultTab();
 }
 
 async function updateUserEventsHosting() {
-  const userEventsHosting = await getUserEvents('hosting');
-  for (const eventsKey in userEventsHosting) {
-    if (userEventsHosting.hasOwnProperty(eventsKey)) {
-      populateEventContainer(userEventsHosting[eventsKey], 'events-hosting', 2);
-    }
-  }
+  await updateUserEvents('hosting');
 }
 
-async function updateUserEventsParticipating() {
-  const userEventsParticipating = await getUserEvents('participating');
-  for (const eventsKey in userEventsParticipating) {
-    if (userEventsParticipating.hasOwnProperty(eventsKey)) {
-      populateEventContainer(userEventsParticipating[eventsKey], 'events-participating', 2);
-    }
-  }
+function updateUserEventsParticipating() {
+  updateUserEvents('participating');
+}
+
+function updateUserEventsVolunteering() {
+  updateUserEvents('volunteering');
 }
 
-async function updateUserEventsVolunteering() {
-  const userEventsVolunteering = await getUserEvents('volunteering');
-  for (const eventsKey in userEventsVolunteering) {
-    if (userEventsVolunteering.hasOwnProperty(eventsKey)) {
-      const eventVolunteering = userEventsVolunteering[eventsKey];
-      eventVolunteering.event.opportunityName = eventVolunteering.opportunityName
-      populateEventContainer(eventVolunteering.event, 'events-volunteering', 3);
+async function updateUserEvents(participatingType) {
+  const userEventsResponse = await getUserEvents(participatingType);
+  let userEvents = [];
+  if (participatingType === 'volunteering') {
+    for (eventMap of userEventsResponse) {
+      userEvents.push(eventMap['event']);
```

Comment:
Here you have removed line: eventVolunteering.event.opportunityName = eventVolunteering.opportunityName but without it it will not show the role for the volunteering opportunities.

Reply:
It has been moved to shared.js - so still will show, looks good.